pymc/distributions/distribution.py

```diff
@@ -622,13 +683,28 @@ def dist(
         if logcdf is None:
             logcdf = default_not_implemented(class_name, "logcdf")
 
+        def dist_moment(rv, size, *dist_params):
+            fgraph = rv.owner.op.fgraph.clone()
```

Comment:
This method can now be used for any OpFromGraph not just CustomDist ones, so I would move it to the MomentRewriter